src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -127,6 +121,11 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
             {
                 resultConstant = null;
             }
+            else if (resultConstant == null)
+            {
+                Debug.Assert(node.Contents.Count == 0);
+                resultConstant = ConstantValue.Create(string.Empty);
+            }
```

Comment:
Consider checking for `node.Contents.Count == 0` early in the method and avoiding `GetSpecialType()` for `System.Int32` and `System.Object`.